File: go/api/jdy/workflow.go
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/5/17
 */

package jdy

import (
	"api-demo/base"
	"api-demo/model"
	"encoding/json"
	"fmt"
)

// WorkflowApiClient 流程Api
type WorkflowApiClient struct {
	*base.ApiClient
}

func (api *WorkflowApiClient) ValidVersion(version string) bool {
	return map[string]bool{
		"v1": true,
	}[version]
}

func (api *WorkflowApiClient) DefaultVersion() string {
	return "v1"
}

// DoRequest 发送http请求
func (api *WorkflowApiClient) DoRequest(option *base.RequestOption) (responseBody []byte, err error) {
	if !api.ValidVersion(api.Version) {
		api.Version = api.DefaultVersion()
	}
	return api.ApiClient.DoRequest(option)
}

// ApprovalComments 获取单条表单流程数据的审批意见
func (api *WorkflowApiClient) ApprovalComments(appId, entryId, dataId string, option map[string]interface{}) (responseBody *model.ApprovalCommentList, err error) {
	payload, err := json.Marshal(option)
	if err != nil {
		return
	}

	response, err := api.DoRequest(&base.RequestOption{
		Method:  "POST",
		Path:    fmt.Sprintf("app/%s/entry/%s/data/%s/approval_comments", appId, entryId, dataId),
		Payload: payload,
	})
	if err != nil {
		return
	}

	err = json.Unmarshal(response, &responseBody)
	return
}

File: python/src/api/jdy/work_flow.py
from ...constants.http_constant import HttpConstant
from ...model.http.http_request_param import HttpRequestParam
from src.model.http.api_client import ApiClient

# 合法的版本
valid_versions = ('v1')
# 默认版本
default_version = 'v1'


class WorkFlowApiClient(ApiClient):

    def __init__(self, api_key, host):
        ApiClient.__init__(self, api_key, host, valid_versions, default_version)

    """
    获取单条表单流程数据的审批意见
    Arguments:
        work_flow_query_param: WorkFlowApprovalCommentQueryParam 实例
        version: 版本
    """

    def approvalComments(self, param, version=default_version):
        url = HttpConstant.APPROVAL_COMMENTS_URL.format(app_id=param.app_id, entry_id=param.entry_id,
                                                        data_id=param.data_id, version=self.getValidVersion(version))
        query_param = {'limit': param.limit, 'skip': param.skip}
        request_param = HttpRequestParam(url, query_param)
        return self.send_post(request_param)
